management_interface/management_interface/forms.py

```diff
@@ -19,9 +25,46 @@ class Meta:
 
 
 class CareRecipientForm(forms.ModelForm):
+    given_name = forms.CharField(max_length=64, required=True, label="Given Name")
+    family_name = forms.CharField(max_length=64, required=True, label="Family Name")
+    nhs_number = forms.CharField(max_length=16, required=True, label="NHS Number", help_text="e.g. 5991993823")
+    birth_date = forms.DateField(required=True, label="Birth Date", help_text="e.g. 1998-01-23")
+
     class Meta:
         model = CareRecipient
         exclude = ["id", "created_at", "updated_at"]
-        labels = {
-            "nhs_number": "NHS Number",
-        }
+
+    def clean(self):
+        subscription_id = self._create_subscription()
+        nhs_number_hash = self._generate_nhs_number_hash()
+        self.cleaned_data["subscription_id"] = subscription_id
+        self.cleaned_data["nhs_number_hash"] = nhs_number_hash
+        self.cleaned_data["nhs_number"] = "".join(self.cleaned_data["nhs_number"].split())
+        return super().clean()
+
+    def save(self, commit: bool = True):
+        self.instance.subscription_id = self.cleaned_data["subscription_id"]
+        self.instance.nhs_number_hash = self.cleaned_data["nhs_number_hash"]
+        return super().save(commit=commit)
+
+    def _create_subscription(self) -> uuid.UUID:
+        try:
+            return ManagementAPIClient().create_subscription(
+                patient_given_name=self.cleaned_data["given_name"],
+                patient_family_name=self.cleaned_data["family_name"],
+                nhs_number=self.cleaned_data["nhs_number"],
+                birth_date=self.cleaned_data["birth_date"],
+            )
+        except ManagementAPIClientError as ex:
+            raise ValidationError(str(ex))
+
+    def _generate_nhs_number_hash(self) -> str:
```

Comment:
Moving this here breaks the bulk upload, right?  What are we doing about that?  [This](https://github.com/nhsx/il-hans-management-interface/blob/main/management_interface/management_interface/admin.py#L84) relies on the model itself calculating the hash.

Reply:
Same: https://github.com/nhsx/il-hans-management-interface/pull/11#discussion_r1156214351